### PR TITLE
[Search] Add reset_documents and reset_skills operations

### PR DESCRIPTION
**Pull request by tjprescott**
These operations were not exposed on the public client.

### CODE REVIEW

File: sdk/search/azure-search-documents/azure/search/documents/indexes/aio/_search_indexer_client.py [hunk 1]
```diff
@@ -278,6 +278,31 @@ async def reset_indexer(self, name, **kwargs):
...
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+        kwargs["headers"] = self._merge_client_headers(kwargs.get("headers"))
+        kwargs["keys_or_ids"] = keys_or_ids
+        try:
+            name = indexer.name
+        except AttributeError:
+            name = indexer
+        result = await self._client.indexers.reset_docs(name, **kwargs)
+        return
```
Comment (xiangyan99) [inline sdk/search/azure-search-documents/azure/search/documents/indexes/aio/_search_indexer_client.py:304]: Do we really need it? (doesn't mypy complain?)

File: sdk/search/azure-search-documents/tests/async_tests/recordings/test_search_index_client_skillset_live_async.test_create_skillset.yaml [hunk 2]
```diff
@@ -1,45 +1,46 @@
 interactions:
 - request:
     body: '{"name": "test-ss", "description": "desc", "skills": [{"@odata.type": "#Microsoft.Skills.Text.EntityRecognitionSkill",
-      "inputs": [{"name": "text", "source": "/document/content"}], "outputs": [{"name":
-      "organizations", "targetName": "organizationsS1"}], "includeTypelessEntities":
-      true}, {"@odata.type": "#Microsoft.Skills.Text.V3.EntityRecognitionSkill", "inputs":
-      [{"name": "text", "source": "/document/content"}], "outputs": [{"name": "organizations",
-      "targetName": "organizationsS2"}], "modelVersion": "3"}, {"@odata.type": "#Microsoft.Skills.Text.SentimentSkill",
+      "name": "skill1", "inputs": [{"name": "text", "source": "/document/content"}],
```
Comment (xiangyan99) [inline sdk/search/azure-search-documents/tests/async_tests/recordings/test_search_index_client_skillset_live_async.test_create_skillset.yaml:4]: Curious why we changed the request payload and added "name"? Will this cause problem?

Comment (xiangyan99) [reply sdk/search/azure-search-documents/tests/async_tests/recordings/test_search_index_client_skillset_live_async.test_create_skillset.yaml:4]: Let me rephrase my question, now we added "name" in the payload, is it required? If yes, why it worked before w/o it? Is it a breaking change?

Comment (tjprescott) [reply sdk/search/azure-search-documents/tests/async_tests/recordings/test_search_index_client_skillset_live_async.test_create_skillset.yaml:4]: You can only reset skills that you have given a name. Hence I had to add the names in the test to be able to test `reset_skills`.

File: sdk/search/azure-search-documents/azure/search/documents/indexes/aio/_search_indexer_client.py (hunk 1 above)
Comment (tjprescott) [reply sdk/search/azure-search-documents/azure/search/documents/indexes/aio/_search_indexer_client.py:304]: Sorry, that should return result.

File: sdk/search/azure-search-documents/tests/async_tests/recordings/test_search_index_client_skillset_live_async.test_create_skillset.yaml (hunk 2 above)
Comment (xiangyan99) [reply sdk/search/azure-search-documents/tests/async_tests/recordings/test_search_index_client_skillset_live_async.test_create_skillset.yaml:4]: To make sure I understand, it is allowed to create a no-named skill. But user will not be able to reset it, right?

Comment (tjprescott) [reply sdk/search/azure-search-documents/tests/async_tests/recordings/test_search_index_client_skillset_live_async.test_create_skillset.yaml:4]: Correct.

**Review by xiangyan99 — APPROVED** [review]
